### DISCUSSION TITLE
Is there a way to influence the merging of transactions when importing multiple accounts?

### DISCUSSION THREAD

**Discussion by goebeler** [firefly-iii/firefly-iii#9280] [Firefly III] [answered]
I checked out Firefly III and want to switch to it from another banking app. The problem is that I want to keep the history of all our family accounts (serveral asset accounts, saving, credit cards).

So I have tried several things to import my data, a little over 9000 transactions, dating back until 2015. I created all accounts before doing the import and created mappings to the accounts, but still many transfers between my accounts are not being merged. 

The result is that the current balance is off by almost 80k€. 

Manually editing the data is just not feasable, it's too much.

In most cases the reason seams to be differences in the descriptions, like one bank appending the used PIN/TAN for example. In other cases it's the date, because the transfer between two banks took a day.

So is there a way I can change the rules how and when transfers are merged?

**Comment by JC5** [comment]
I assume you're importing data using CSV files, correct?

**Reply by goebeler** [reply]
Yes

**Reply by goebeler** [reply]
The csv files are an export from Subsembly's Banking4 App. I tried both, using individual files, one per account or creating a big merged one. It did not really make a difference. By the way, the importer displays a timeout error message, both during conversion and during the actual import, but the process continues in the background. But that's a different topic and probably has nothing to do with my actual problem.

**Reply by JC5** [reply]
Thanks! Does the CSV file have a column with a unique ID, like an identifier or something?

**Reply by goebeler** [reply]
Yes. But the are different/unique for both transactions.

**Reply by goebeler** [reply]
I think I could try to remove one direction of every transfer from the dataset, this should be pretty easy to automate. I could try to use a db or even excel to remove every transaction which has one of my accounts as source and one of the other accounts as destination. That should leave me with only one direction of every transfer right? 

Do you know a better solution?

**Reply by JC5** [reply, marked as answer]
That would work, absolutely.

Try importing one month at a time to find out any bugs in your code, and progress slowly.
